src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/AbstractAppSchemaMockData.java

```diff
@@ -42,6 +42,8 @@
 public abstract class AbstractAppSchemaMockData extends SystemTestData
         implements NamespaceTestData {
 
+    private String layerNamePrefix;
```

Comment:
See my comments bellow, but long story short, this behavior should not be implement here, it should be up to the test setup or test using this class to set the final type name.

If everyone starts adding its specific bit of business to this class it would end up with 10'000  lines.